scheduler/scd_utils.py

```diff
@@ -1,100 +1,142 @@
+        :type   scheduling_mode:    str
+        :param  prio:               priority value.
+        :type   prio:               str or int
+        :param  duration:           an optional duration to run for.
+        :type   duration:           str
+        :param  kwargs_string:      kwargs for the experiment instantiation. Default None
+        :type   kwargs_string:      str
+
+        :returns:   Dict of line params.
+        :rtype:     dict
+
+        :raises     ValueError: If line parameters are invalid or if line is a duplicate.
         """
 
-        try:
-            # create datetime from args to see if valid
-            time = dt.datetime.strptime(yyyymmdd + " " + hhmm, self.scd_dt_fmt)
-        except:
-            raise ValueError("Can not create datetime from supplied formats")
-        try:
-            int(prio)
-        except ValueError as e:
-            raise ValueError("Unable to cast priority {} as int.".format(prio))
+        # create datetime from args to see if valid. Value error for incorrect format
+        time = dt.datetime.strptime(yyyymmdd + " " + hhmm, self.scd_dt_fmt)
+
+        if not isinstance(kwargs_string, str):
+            raise ValueError("kwargs_string should be a string")
 
         if not (0 <= int(prio) <= 20):
             raise ValueError("Priority is out of bounds. 0 <= prio <= 20.")
 
-
         if duration != "-":
-            try:
-                int(duration)
-            except ValueError as e:
-                raise ValueError("Unable to cast duration {} as int".format(duration))
+            if isinstance(duration, float) or int(duration) < 1:
+                raise ValueError("Duration should be an integer > 0, or '-'")
+            duration = int(duration)
 
         epoch = dt.datetime.utcfromtimestamp(0)
         epoch_milliseconds = int((time - epoch).total_seconds() * 1000)
 
         possible_scheduling_modes = ['common', 'special', 'discretionary']
         if scheduling_mode not in possible_scheduling_modes:
-            raise ValueError("Unknown scheduling mode type {} not in {}"
-                .format(scheduling_mode, possible_scheduling_modes))
-
-        return {"timestamp" : epoch_milliseconds,
-                "time" : time,
-                "duration" : str(duration),
-                "prio" : str(prio),
-                "experiment" : experiment,
-                "scheduling_mode" : scheduling_mode,
+            raise ValueError(f"Unknown scheduling mode type {scheduling_mode} not in {possible_scheduling_modes}")
+
+        # Don't bother testing past experiments, formats/settings/capabilities/etc. could have changed
+        if not dt.datetime.utcnow() - time > dt.timedelta(days=1):  # Test if experiment is in future or past day
+            # See if the experiment itself would run
+            # This is a full path to /.../{site}.scd file, only want {site}
+            site_name = os.path.basename(self.scd_filename).replace('.scd', '')
```

Comment:
Could get this from remote_server_options.py (or something similar, instead of using .scd file name)